ibis/expr/datatypes.py

```diff
@@ -1232,6 +1235,17 @@ def can_cast_arrays(source, target, **kwargs):
     return castable(source.value_type, target.value_type)
 
 
+@castable.register(Struct, Struct)
+def can_cast_struct(source, target, **kwargs):
+    return (
+        (source.names <= target.names) and
```

Comment:
Don't need parens here.

Reply:
Is `names` a set?

Reply:
Are the semantics: "`source` can be cast to `target` if the names of `source` are a subset or equal to the names of `target` and types corresponding to each of those names in each struct type are equivalent"?

Reply:
names is of type `odict_keys`
re semantics: yes; re subset: perhaps there should be a check on the nullability of the missing fields?

Reply:
I don't think this will work in python 2 because keys are a list IIRC